eng/versioning.targets

```diff
@@ -40,17 +40,24 @@
   <ItemGroup Condition="('$(TargetsBrowser)' == 'true' or '$(CrossPlatformAndHasNoBrowserTarget)' == 'true') and '$(IsTestProject)' != 'true'">
     <SupportedPlatform Include="browser"/>
   </ItemGroup>
-
-  <!-- Enables warnings for tvOS targeted builds -->
-  <ItemGroup Condition="'$(TargetstvOS)' == 'true' and '$(IsTestProject)' != 'true'">
-    <SupportedPlatform Include="tvos"/>
+  
+  <!-- Add target platforms into MSBuild SupportedPlatform list -->
+  <ItemGroup Condition="'$(TargetsAnyOS)' != 'true' and '$(IsTestProject)' != 'true'">
+    <SupportedPlatform Condition="'$(Targetsillumos)' == 'true'" Include="illumos"/>
+    <SupportedPlatform Condition="'$(TargetsMacCatalyst)' == 'true'" Include="MacCatalyst"/>
+    <SupportedPlatform Condition="'$(TargetsSolaris)' == 'true'" Include="Solaris"/>
+    <SupportedPlatform Condition="'$(TargetstvOS)' == 'true'" Include="tvos"/>
+    <SupportedPlatform Condition="'$(TargetsUnix)' == 'true'" Include="Unix"/>
   </ItemGroup>
   
   <ItemGroup>
     <_unsupportedOSPlatforms Include="$(UnsupportedOSPlatforms)" />
   </ItemGroup>
 
-  <Target Name="AddUnsupportedOSPlatformAttribute" BeforeTargets="GenerateAssemblyInfo" AfterTargets="PrepareForBuild" Condition="'@(_unsupportedOSPlatforms)' != '' and '$(IsTestProject)' != 'true' and '$(IncludePlatformAttributes)' != 'false'">
+  <!-- Adds UnsupportedOSPlatform attribute to the assembly when the project has non empty <UnsupportedOSPlatforms/> property and
+       it is not a test project and not targeting .Net framework and PlatformAttributes available and having cross-platform target -->
```

Comment:
```suggestion
  <!-- Adds UnsupportedOSPlatform attribute to the assembly when:
      * At least one <UnsupportedOSPlatforms /> has been specified
      * This isn't a test project
      * The PlatformAttributes are available
      * This is a cross-platform target
      * The build isn't targeting .NET Framework
    -->
```

@safern @ViktorHofer -- Do you see any opportunities to simplify the condition here?

Reply:
We might not need to check  `The PlatformAttributes are available` e.g. `'$(IncludePlatformAttributes)' != 'false'` condition, i don't think we ever set `IncludePlatformAttributes` to `false`, only set it to `true` to have the attributes in < 5.0  versions. So that condition will always be true, even if attributes not exist, there will be a build error instead if the attributes not found

Reply:
Updated the comment, removed the `'$(IncludePlatformAttributes)' != 'false'` condition